src/effects/Reverb_libSoX.h

```diff
@@ -218,6 +268,174 @@ typedef struct {
+   //  | B A |          smaller size, and ptr would be beyond the new right edge: left-shift what is ahead of ptr,
+   //                   AND move ptr to size-1
+
+   assert(newSize > 0);
+
+   const int    sizeDiff = (newSize - p->size);
+   const size_t ptrPos   = (p->ptr  - p->buffer);
+
+   const size_t numSamplesBehindPtr = (p->size - 1) - ptrPos;
+
+   if (sizeDiff > 0)
+   {
+      // case: bigger size
+
+      if (numSamplesBehindPtr > 0)
+      {
+         // right-shift what is right of ptr
+         memcpy(p->ptr + 1 + sizeDiff, p->ptr + 1, numSamplesBehindPtr * sizeof(float));
+      }      
+
+      // fill the created gap with zeros
+      memset(p->ptr + 1, 0, sizeDiff * sizeof(float));
+   }
+   else if (sizeDiff < 0)
+   {
+      // case: smaller size
+            
+      if (ptrPos < newSize-1)
+      {
+         size_t lenOfBlockToShift = newSize - 1 - ptrPos;
+         float* ptrToBlockToShift = p->buffer + p->size - lenOfBlockToShift;
+         memcpy(p->ptr + 1, ptrToBlockToShift, lenOfBlockToShift * sizeof(float));         
+      }
+      else if (ptrPos == newSize - 1)
+      {
+         // sub-case: ptr is at the new edge - no shifting to do, and ptr can stay where it is
+      }
+      else
+      {
+         // sub-case: ptr would be beyond the new edge
+         // left-shift what is ahead of ptr and make ptr point to the new edge
+         size_t samplesBeyond = ptrPos - newSize + 1;
+         memcpy(p->buffer, p->ptr - newSize + 1, newSize * sizeof(float));
+         p->ptr = p->buffer + newSize - 1;
+      }
+   }
+
+   p->size = newSize;
+}
+
+
+static void filter_t_resize(filter_t* p, size_t newSize)
+{
+   // Choose your resize method.
+   //
+   // When moving the Room Size slider:
+   //
+   // - using resize_resetting, you will not hear previous reverberations,
+   //   but at the same time you will not hear sound artifacts
+   //
+   // - using resize_preserving, you will hear previous reverberations,
+   //   but sometimes you might hear sound artifacts
+
+#if 1
+   filter_t_resize_resetting(p, newSize);
+#else
+   filter_t_resize_preserving(p, newSize);
+#endif
+
+}
+
+
+static void reverb_allocate(reverb_t* p, double sample_rate_Hz, size_t buffer_size, float** out)
+{
+   memset(p, 0, sizeof(*p));
+
+   // Input queue
+   fifo_create(&p->input_fifo, sizeof(float));
+
+   // Outputs
+   out[0] = lsx_zalloc(p->out[0], buffer_size);
+   out[1] = lsx_zalloc(p->out[1], buffer_size);
+
+   // Allpass & Comb filters
+   filter_array_allocate(p->chan + 0, sample_rate_Hz, 1.0, 0.0);
+   filter_array_allocate(p->chan + 1, sample_rate_Hz, 1.0, 1.0);
+}
+
+static void reverb_init
+(
+   reverb_t* p,
+   double sample_rate_Hz,
+   double wet_gain_dB,
+   double room_scale,     /* % */
+   double reverberance,
+   double hf_damping,
+   double pre_delay_ms,
+   double stereo_depth,
+   double tone_low,       /* % */
+   double tone_high       /* % */
+)
+{
+   // Input queue
+   p->input_fifo.allocation = FIFO_MIN;
```

Comment:
Is it correct to clear the FIFO whenever you reinitialize?

Or should this be done only when you also use `filter_t_resize_resetting`?

Reply:
Or maybe instead, write a new function that overwrites the fifo with zeroes.

Reply:
Yes, I found out that if I did not do fifo_clear when I reinitialize, then the output would not sound right (specifically: when dragging the room size slider from 100% to 0% while playing, what I would get would not sound as if I would have started the reverb at 0% from scratch)

And it is true, there is no need to also set .allocation to FIFO_MIN - not doing that, can avoid later allocations. I changed that.